contracts/abstracts/TokenCollector.sol

```diff
@@ -53,12 +52,11 @@ abstract contract TokenCollector {
             }
             return IERC20(token).safeTransferFrom(from, to, amount);
         } else if (src == Source.Permit2AllowanceTransfer) {
-            if (amount > uint256(type(uint160).max)) revert Permit2AmountTooLarge();
             bytes memory permit2Data = data[1:];
             if (permit2Data.length > 0) {
                 (uint48 nonce, uint48 deadline, bytes memory permitSig) = abi.decode(permit2Data, (uint48, uint48, bytes));
                 IUniswapPermit2.PermitSingle memory permit = IUniswapPermit2.PermitSingle({
-                    details: IUniswapPermit2.PermitDetails({ token: token, amount: uint160(amount), expiration: deadline, nonce: nonce }),
+                    details: IUniswapPermit2.PermitDetails({ token: token, amount: type(uint160).max, expiration: deadline, nonce: nonce }),
```

Comment:
I should have leave the comment above here instead. Sorry for the confusion. So what I meant is that maybe we don't have to enforce amount being max here? We still permit with whatever `amount` that are specified by the user and we can probably check that off-chain.

Reply:
Currently the `amount` is used for both transfer amount and approval amount. So it indeed hard to fulfill cases like transferring specific amount but approving max amount.

Another idea is that maybe we could just pass through arbitrary data to uniswap permit, just like the token permit case, to allow users to customize their needs.

Me either thinking that hardcoding `max` is not that flexible as an on-chain protocol.

Reply:
I think the arbitrary data with `IUniswapPermit2.permit.selector` should be able to work.

Reply:
Fixed in abfe3e1.